docs/content/en/docs/Design/relinearization_ilp.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Optimizing relinearization
+weight: 10
+---
+
+This document outlines the integer linear program model used in the
+[`optimize-relinearization`](https://heir.dev/docs/passes/optimizerelinearizationpasses/#-optimize-relinearization)
+pass.
+
+## Background
+
+In vector/arithmetic FHE, RLWE ciphertexts often have the form $\\mathbf{c} =
+(c_0, c_1)$, where the details of how $c_0$ and $c_1$ are computed depend on the
+specific scheme. However, in most of these schemes, the process of decryption
+can be thought of as taking a dot product between the vector $\\mathbf{c}$ and a
+vector $(1, s)$ containing the secret key $s$ (followed by rounding).
+
+In such schemes, the homomorphic multiplication of two ciphertexts $\\mathbf{c}
+= (c_0, c_1)$ and $\\mathbf{d} = (d_0, d_1)$ produces a ciphertext $\\mathbf{f}
+= (f_0, f_1, f_2)$. This triple can be decrypted by taking a dot product with
+$(1, s, s^2)$.
+
+With this in mind, each RLWE ciphertext $\\mathbf{c}$ has an associated _key
+basis_, which is the vector $\\mathbf{s_c}$ whose dot product with $\\mathbf{c}$
+decrypts it.
+
+Usually a larger key basis is undesirable. For one, operations in a higher key
+basis are more expensive. Repeated multiplications exponentially increase the
```

Comment:
It's probably worth mentioning that it also leads to really bad noise growth behavior.

Reply:
Fixed in eac36665